pkg/kms/service.go

```diff
@@ -67,3 +74,34 @@ func (s *service) CreateKey(keystoreID string, kt kms.KeyType) (string, error) {
 
 	return keyID, nil
 }
+
+// Sign signs a message using the key identified by keyID.
+func (s *service) Sign(keystoreID, keyID string, msg []byte) ([]byte, error) {
+	k, err := s.keystore.Get(keystoreID)
+	if err != nil {
+		return nil, fmt.Errorf(getKeystoreErr, err)
+	}
+
+	if len(k.KeyIDs) == 0 {
```

Comment:
strange condition... are we allowing keystores without keys? Looking at `CreateKey` above, seems not...

Reply:
it can be a case
initially you create a keystore, then a key, so it's possible that Sign is called on "empty" keystore